lib/client.js

```diff
@@ -285,6 +390,43 @@ function addToBuffer(self, metricLines, isMetricAggregated, agg, aggFreq) {
 }
 
 /**
+ * Adds raw metrics directly into the flush buffer. Use this method with caution.
+ *
+ * @param self A self client instance.
+ * @param metricLines The metrics, in valid line protocol, to push to the buffer.
+ * @param isMetricAggregated A boolean state if metrics are aggregated.
+ * @param agg In case that metric is aggregated we need to send the aggregation.
+ * @param aggFreq In case that metric is aggregated we need to send the aggregation frequency.
+ */
+function addToStatsBuffer(self, metricLines, isMetricAggregated, agg, aggFreq) {
+
+    if (typeof metricLines !== 'undefined') {
+        var targetBuffer = isMetricAggregated ? self.aggregatedStatsBuffer : self.nonAggregatedStatsBuffer;
+
+        if (isMetricAggregated) {
+            if (targetBuffer[agg][aggFreq].buffer.length > 0) {
+                targetBuffer[agg][aggFreq].buffer += '\n';
+            }
+
+            targetBuffer[agg][aggFreq].buffer += metricLines;
+            targetBuffer.bufferSize++;
+        } else {
+            if (targetBuffer.bufferSize > 0) {
+                targetBuffer.buffer += '\n';
+            }
+
+            targetBuffer.buffer += metricLines;
+            targetBuffer.bufferSize++;
+        }
+
```

Comment:
Well, that is exactly the reason we are doing this fix. When we call the flush function if the buffer is "full" (independent of the reason/threshold), it will call again the function to put the stats metrics in the buffer and so on and so on... I'd rather have another interval (smaller than the timeout to flush the other metrics) to flush the stats metrics.